### PR TITLE
Rework test setup

### PR DESCRIPTION
**Pull request by jomey**
Went through the test suite and:
* Changed the test credential file setup to be able to customize based on personal environment
* Cleaned up unused logic
* Created smaller test files
* Consolidated the logic on DB handling
* Added instructions to the docs to get set up
* Added 3.11 and 3.12

Note: 3.8 is EOL in a month (https://endoflife.date/python)

### CODE REVIEW

**Comment by jomey** [comment]
Tests are not running here since I am not targeting the release branch. Could also wait until #135 is merged and change the target

**Comment by jomey** [comment]
@micah-prime - Not sure why a couple of your commits are showing up here. I basically went from the HEAD of your `issue_133_134` branch

File: .github/workflows/main.yml
Comment (micah-prime) [inline .github/workflows/main.yml:21]: Hey github must have gotten better at parsing these versions!

**Comment by micah-prime** [comment]
> @micah-prime - Not sure why a couple of your commits are showing up here. I basically went from the HEAD of your `issue_133_134` branch

Probably just some squashing, not a big deal

**Review by micah-prime — APPROVED** [review]
Very nice!

**Pull request merged**
